examples/plot_clip.py

```diff
@@ -0,0 +1,135 @@
+"""
+Clip Vector Data with GeoPandas
+==================================================================
+
+Learn how to clip point, line, or polygon geometries to the boundary
+of a polygon geometry using GeoPandas.
+
+"""
+
+###############################################################################
+# Clip Vector Data in Python Using GeoPandas
+# ---------------------------------------------
+#
+# The example below shows you how to clip a set of vector geometries
+# to the spatial extent / shape of another vector object. Both sets of geometries
+# must be opened with GeoPandas as GeoDataFrames and be in the same Coordinate
+# Reference System (CRS) for the ``clip()`` function in EarthPy to work.
+#
+# This example uses Polygons, a Line, and Points made with shapely and then
+# turned into GeoDataframes.
+#
+# .. note::
+#    The object to be clipped will be clipped to the full extent of the clip
+#    object. If there are multiple polygons in clip object, the input data will
+#    be clipped to the total boundary of all polygons in clip object.
+
+###############################################################################
+# Import Packages
+# ---------------
+#
+# To begin, import the needed packages.
+
+import numpy as np
+import matplotlib.pyplot as plt
+import geopandas as gpd
+from shapely.geometry import Polygon, LineString, Point
+import geopandas.clip as gc
+
+###############################################################################
+# Create Example Data
+# -------------------
+#
+# Below, some point, line and polygon geometries are created and coerced into
+# GeoDataFrames to demonstrate the use of clip.
+
+polygon1 = Polygon([(0, 0), (0, 10), (10, 10), (10, 0), (0, 0)])
+polygon2 = Polygon([(-5, -5), (-5, 5), (5, 5), (5, -5), (-5, -5)])
+line = LineString([(3, 4), (5, 7), (12, 2), (10, 5), (9, 7.5)])
+pts = np.array([[2, 2], [3, 4], [9, 8]])
```

Comment:
I was wondering, maybe the example could be more "interesting" with more real-world data? 
We don't have much data included in geopandas, but in principle we could (again) use the countries and cities datasets. The countries could be clipped with a box, and then eg the cities could be clipped with a subset of the countries datasets?

Reply:
(but this is certainly not high priority, it's already great there is an example!)

Reply:
Hey @jorisvandenbossche ! I decided to use custom geometries in this vignette to showcase it working with lines as well as points and polygons. I can't really think of a way to show line clipping functionality with the GeoPandas datasets. If you are ok with no lines examples (which it sounds like you are!) I could change it to use the GeoPandas data like you suggested.

Reply:
What about clipping `.exterior` of countries by large buffer around cities? Which leads me to the point - clip should support LinearRings as well, it does not now.

Reply:
@martinfleis i propose that we focus on getting the functionality in clip now working the way you'd like it and merge this PR. then we can add additional support as a separate PR? would you and @jorisvandenbossche be amenable to this? Or if you'd like to add to this PR that specific functionality, that would work as well.

Reply:
@lwasser this is just a minimal change, I just proposed it in the code, you can just commit it. Agree with the rest. Finalise this, we'll merge it and then we can think about other improvements (but I think that is is quite versatile already).

Reply:
thank you so much @martinfleis !! i'm jumping back and forth between too many projects today and didn't realize it was a small change! I appreciate that commit! thank you.

i'm also going through now to ensure we didn't miss anything else. i'll resolve comments to make it a bit easier to track :)